timesketch/lib/analyzers/sigma_tagger.py

```diff
@@ -71,64 +46,54 @@ def run(self):
-                            'this is a directory, skipping: {0:s}'.format(
-                                rule_filename))
-                        continue
-
-                    tag_name, _ = rule_filename.rsplit('.')
-                    tags_applied[tag_name] = 0
-                    rule_file_path = os.path.join(dirpath, rule_filename)
-                    rule_file_path = os.path.abspath(rule_file_path)
-                    logger.info('[sigma] Reading rules from {0!s}'.format(
-                        rule_file_path))
-                    with codecs.open(rule_file_path, 'r', encoding='utf-8',
-                                     errors='replace') as rule_file:
-                        try:
-                            rule_file_content = rule_file.read()
-                            parser = sigma_collection.SigmaCollectionParser(
-                                rule_file_content, self.sigma_config, None)
-                            parsed_sigma_rules = parser.generate(sigma_backend)
-                        except NotImplementedError as exception:
-                            logger.error(
-                                'Error generating rule in file {0:s}: {1!s}'
-                                .format(rule_file_path, exception))
-                            continue
-
-                        for sigma_rule in parsed_sigma_rules:
-                            try:
-                                sigma_rule_counter += 1
-                                logger.info(
-                                    '[sigma] Generated query {0:s}'
-                                    .format(sigma_rule))
-                                tagged_events_counter = self.run_sigma_rule(
-                                    sigma_rule, tag_name)
-                                tags_applied[tag_name] += tagged_events_counter
-                            except elasticsearch.TransportError as e:
-                                logger.error(
-                                    'Timeout generating rule in file {0:s}: '
-                                    '{1!s} waiting for 10 seconds'.format(
-                                        rule_file_path, e), exc_info=True)
-                                time.sleep(10) # waiting 10 seconds
+        try:
+            rules_path = ts_sigma_lib.get_sigma_rules_path()
+
+        except ValueError:
+            logger.error('OS Error, unable to get the path to the Sigma rules',
+                         exc_info=True)
+            return False
+
+        sigma_rules = ts_sigma_lib.get_sigma_rules(rules_path)
+
+        if sigma_rules is None:
+            logger.error('error rule is none')
+
+        problem_string = 'Problematic rules:\n'
+
+        for rule in sigma_rules:
+            if rule is None:
+                logger.error('error rule is none')
+                continue
+            tags_applied[rule.get('file_name')] = 0
+            logger.debug("Rule : {0:s} with query {1:s}".format(
```

Comment:
single quote vs double

Reply:
and is this something we want to keep in here... this sounds like something that would generate a lot of debug entries in the logs, and not sure how helpful (except when you are developing this)

Reply:
true, removed it, the except cases should be good enough in case something goes wrong to tell